### ISSUE TITLE
App stops detecting SMS

### ISSUE THREAD

**Issue by bilogic** [OptimalStrategy/sms_forwarder_app#16]
Hi,

The app recently started to stop detecting SMSes after running for a while, i.e. SMS was received by phone's SMS app, but no action taken by app. I tried SMS Forwarder's `Send Test Message` and it works.

Any idea how to figure out why this happens? Thank you.

**Comment by optimalstrategy** [comment]
The app registers an [SMS receiver](https://github.com/optimalstrategy/sms_forwarder_app/blob/11d873c4331203a493c20f4c78d64c779c38a5b9/android/app/src/main/AndroidManifest.xml#L43) that should always listen for new messages in the background, even when you close the app. I suspect that it might be getting killed for some reason -- could you please check if background execution is enabled for the app? On my device this setting is located in SMS Forwarder -> App Settings -> Battery -> Allow Background Activity.

I have also done a clean install on one of my spare phones; I'll leave it running for the next few days and see if I can reproduce the issue.

**Comment by bilogic** [comment]
Hi! 

Thanks for getting back. Yes, `Allow Background Activity` is enabled for SMS Forwarder.

When it did not work earlier, I killed both the SMS app and SMS Forwarder, then restarted SMS Forwarder first.
That still did not solve the problem, but I did not pursue further.

Upon reading your email now, and making the battery check, I decided to test it again, this time, it forwarded an SMS.
I will monitor further and see there is a consistent way to reproduce it.

Thank you.
